e2e/pull/pull.go

```diff
@@ -323,68 +341,51 @@ func (c *ctx) testPullCmd(t *testing.T) {
 				os.Chdir(tmpdir)
 
 				// if there's a pullDir, that's where we expect to find the image
-				if pullDir != "" {
-					os.Chdir(pullDir)
+				if tt.pullDir != "" {
+					os.Chdir(tt.pullDir)
 				}
 
-				expectedImage = getImageNameFromURI(tt.srcURI)
+				tt.expectedImage = getImageNameFromURI(tt.srcURI)
 			}
 
 			// In order to actually test force, there must already be a file present in
 			// the expected location
 			if tt.createDst {
-				fh, err := os.Create(expectedImage)
+				fh, err := os.Create(tt.expectedImage)
 				if err != nil {
-					t.Fatalf("Failed to create file %q: %+v\n", expectedImage, err)
+					t.Fatalf("failed to create file %q: %+v\n", tt.expectedImage, err)
 				}
 				fh.Close()
 			}
 
-			cmd, out, err := c.imagePull(t, tt.srcURI, tt.library, pullDir, imagePath, tt.force, tt.unauthenticated)
-			switch {
-			case tt.expectSuccess && err == nil:
-				// MAYBE PASS: expecting success, succeeded
-
-				_, err := os.Stat(expectedImage)
-				switch err {
-				case nil:
-					// PASS
-					return
-
-				case os.ErrNotExist:
-					// FAIL
-					t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-					t.Errorf("expecting image at %q, not found: %+v\n", expectedImage, err)
-
-				default:
-					// FAIL
-					t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-					t.Errorf("unable to stat image at %q: %+v\n", expectedImage, err)
-				}
-
-				// XXX(mem): This is running a bunch of commands in the downloaded
-				// images. Do we really want this here? If yes, we need to have a
-				// way to do this in a generic fashion, as it's going to be shared
-				// with build as well.
-
-				// imageVerify(t, tt.imagePath, false)
-
-			case !tt.expectSuccess && err != nil:
-				// PASS: expecting failure, failed
-
-			case tt.expectSuccess && err != nil:
-				// FAIL: expecting success, failed
+			c.imagePull(t, tt)
+		})
+	}
+}
 
-				t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-				t.Errorf("unexpected failure: %v", err)
+func checkPullResult(t *testing.T, tt testStruct) {
+	if tt.expectedExitCode == 0 {
+		_, err := os.Stat(tt.expectedImage)
+		switch err {
+		case nil:
+			// PASS
+			return
+
+		case os.ErrNotExist:
+			// FAIL
+			t.Errorf("expecting image at %q, not found: %+v\n", tt.expectedImage, err)
+
+		default:
+			// FAIL
+			t.Errorf("unable to stat image at %q: %+v\n", tt.expectedImage, err)
+		}
 
-			case !tt.expectSuccess && err == nil:
-				// FAIL: expecting failure, succeeded
+		// XXX(mem): This is running a bunch of commands in the downloaded
+		// images. Do we really want this here? If yes, we need to have a
+		// way to do this in a generic fashion, as it's going to be shared
+		// with build as well.
 
-				t.Logf("Running command:\n%s\nOutput:\n%s\n", cmd, out)
-				t.Errorf("unexpected success: command should have failed")
-			}
-		})
+		// imageVerify(t, tt.imagePath, false)
```

Comment:
is this a `TODO`? might be needed to be removed in any case if not..

Reply:
No idea, this is not my code :)